lib/nrf/51/i2c.c

```diff
@@ -0,0 +1,214 @@
+/** @addtogroup i2c_defines
+ *
+ * @brief <b>Access functions for the NRF51 I2C </b>
+ * @ingroup NRF51_defines
+ * LGPL License Terms @ref lgpl_license
+ * @author @htmlonly &copy; @endhtmlonly 2016
+ * Maxim Sloyko <maxims@google.com>
+ *
+ */
+
+/*
+ * This file is part of the unicore-mx project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <unicore-mx/nrf/i2c.h>
+
+/** @brief Enable I2C peripheral
+ *
+ * @param[in] i2c uint32_t i2c peripheral base
+ */
+void i2c_enable(uint32_t i2c)
+{
+	I2C_ENABLE(i2c) = I2C_ENABLE_VALUE;
+}
+
+/** @brief Disable I2C peripheral
+ *
+ * @param[in] i2c uint32_t i2c peripheral base
+ */
+void i2c_disable(uint32_t i2c)
+{
+	I2C_ENABLE(i2c) = 0;
```

Comment:
@brabo The Common Peripheral Interface is ready for review -- just in case you missed that :) Once that PR gets in, I'll update this one too to use that interface.

Thank you!